pkg/karmadactl/cmdinit/karmada/deploy.go

```diff
@@ -28,7 +28,7 @@ import (
 const namespace = "karmada-system"
 
 // InitKarmadaResources Initialize karmada resource
-func InitKarmadaResources(dir, caBase64 string) error {
+func InitKarmadaResources(dir, caBase64 string, externalNamespace string) error {
```

Comment:
```suggestion
func InitKarmadaResources(dir, caBase64 string, systemNamespace string) error {
```
I suggest renaming the parameter name to `systemNamespace`, because there is no concept of `external` or `internal` here.

Reply:
https://github.com/karmada-io/karmada/blob/867160a2f219c1a374c209cba011edf9a9d74dea/pkg/karmadactl/cmdinit/karmada/deploy.go#L176-L179

It's just like the external service above.

https://github.com/karmada-io/karmada/blob/867160a2f219c1a374c209cba011edf9a9d74dea/pkg/karmadactl/cmdinit/karmada/deploy.go#L80-L95

It's the namespace out of Karmada, which is used for webhook and karmada-aggregated-apiserver in Karmada.

The `external` and `internal` are for Karmada.

Reply:
I don't think so. 
The namespace is used to specify the location of the `service`(karmada-webhook, karmada-aggregate-apiserver) which is supposed in the Karmada system namespace by default.

Reply:
https://github.com/karmada-io/karmada/blob/867160a2f219c1a374c209cba011edf9a9d74dea/pkg/karmadactl/cmdinit/karmada/deploy.go#L167-L180

The namespace is `karmada-system` in Karmada which is hard coded.

The externalNamespace is custom namespace in host cluster where Karmada is deloyed.

Reply:
Yeah, I get what you mean. Once user specified the customized namespace by `--namespace`, like `myns`, which means:
- Installing Karmada components to the `myns` namespace on `host` cluster. (As well as the services...)
- Karmada will use the `myns` as the system namespace to store some configuration resources, like `service/karmada-aggregated-apiserver`.

Are you agree with that? Personally, I don't like to identify the namespace by `internal` or `external` as I think it brings a bit of confusion.

Reply:
OK, I'll make the change later.

Reply:
Yeah, you are right. Seems we have a lot of things to do.